### PR TITLE
Single cell route improvements and enhanced cohort level filtering with other minor changes

### PR DESCRIPTION
**Pull request by creilly8**
# Description
This pull request introduces new integration tests for single-cell chart types, refactors how subplot parent IDs are handled, improves metadata handling for single-cell subplots, and reorganizes server route imports for single-cell functionality. It also includes bug fixes and test coverage improvements for edge cases in special case handling. Feedback from [previous PR](https://github.com/stjude/proteinpaint/pull/4600) was also addressed. This PR supports the ongoing efforts in the [SC roadmap](https://github.com/stjude/sjpp/issues/658). 

Changes: 
1. Main change -> Reorganized server route imports to move single-cell routes (`singlecellSamples`, `singleCellPlots`, etc.) into a new `singleCell` folder and updated all relevant imports and route lists for consistency and maintainability. These include our new route 'checkers' in the route file as well as request and response examples. 
2. Created a new method to retrieve filtered sample based on `mayLimitSamples`. This solved the problem of the MMRF samples not filtering since a termdb is not used. 
3. getData() in termdb.matrix now accounts for filtering samples when only single cell terms are used in meta results plots, specifically filter0. Nested filters with cohort level terms from the parent (SC) and local summary plot appropriately filter out cells. 
4. Refactored subplot initialization to consistently propagate the root `parentId` through summary and dynamic subplots, ensuring correct parent-child relationships for nested single-cell subplots.
5. Improved metadata result handling by updating how `isMetaResult` is set for items, moving this logic from the renderer to the `SCInteractions` class and using `viewModel.metaResultIds` for determination.

Minor changes:
1. 'cells' is returned as the sampleType when single cell terms are processed in getData(). Note: this is a workaround since all the look up methods use either the ds termdb or api calls. This fixes the problem with 'samples' instead of 'cells' appearing in the barchart title. 
2. Updated `getSingleCellSpecialCase` to handle missing or invalid term objects gracefully, with additional tests for these edge cases. 
3. Updated filter handling in `SCModel` to use `getNormalRoot`, ensuring correct filter structure for single-cell sample queries.
4. Added integration tests for single-cell chart types, including tests for boxplots and barcharts using `scct` and `scge` terms in `boxplot.integration.spec.ts` and `barchart.integration.spec.js`. 
5. Corrected documentation in `README.md` for route file creation instructions.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.

- [x] Tests: Added and/or passed unit and integration tests, or N/A
- [x] Todos: Commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A
- [x] Rust: Checked to see whether Rust needs to be re-compiled because of this PR, or N/A

### CODE REVIEW

File: client/plots/sc/model/SCModel.ts [hunk 1]
```diff
@@ -20,8 +21,8 @@ export class SCModel {
 		const body = {
 			genome: state.vocab.genome,
 			dslabel: state.vocab.dslabel,
-			filter: state.termfilter.filter || null,
-			filter0: state.termfilter.filter0 || null
+			filter: getNormalRoot(state.termfilter.filter),
+			filter0: getNormalRoot(state.termfilter.filter0)
 		}
```
[the remark this answers is not in the document]
Comment (xzhou82) [reply client/plots/sc/model/SCModel.ts:26]: cannot run getNormalRoot on filter0

**Review by xzhou82 — APPROVED** [review]
thanks
